addons/hw_drivers/tools/helpers.py

```diff
@@ -388,12 +421,21 @@ def load_iot_handlers():
                     _logger.exception('Unable to load handler file: %s', file)
     lazy_property.reset_all(http.root)
 
-def list_file_by_os(file_list):
-    platform_os = platform.system()
-    if platform_os == 'Linux':
-        return [x.name for x in Path(file_list).glob('*[!W].*')]
-    elif platform_os == 'Windows':
-        return [x.name for x in Path(file_list).glob('*[!L].*')]
+def list_file_by_iot_system(file_list: str) -> list:
+    """
+    List all handler files that an IoT system should load.
+    - IoT boxes load file without suffixe and _L
+    - Windows IoT load file without suffixes and _W
+    - Test IoT only load files with _T suffix
+    :param file_list: The path to the directory containing the files (either drivers or interfaces)
+    """
```

Comment:
```suggestion
def list_file_by_iot_system(file_list):
    """
    List all handler files that an IoT system should load.
    - IoT boxes load file without suffixe and _L
    - Windows IoT load file without suffixes and _W
    - Test IoT only load files with _T suffix
    
    :param str file_list: The path to the directory containing the files (either drivers or interfaces)
    :return: files corresponding to the current IoT system
    :rtype list:
    """
```
It was decided that we don't type variables, you can add types in the method comment.

Reply:
I really think type hinting could be useful, but ok

Reply:
> It was decided that we don't type variables, you can add types in the method comment.

I didn't know this, was this an IoT specific thing?
Personally I'd support us having type hints, it's always a plus IMO

Reply:
Yes I would support us having type hints as well, but I already debated it with @Ysoroko: if i remember well, it is a decision odoo-wide